Framework/CurveFitting/src/Functions/CrystalFieldFunction.cpp

```diff
@@ -128,9 +128,6 @@ size_t CrystalFieldFunction::getNumberDomains() const {
   if (!m_target) {
     buildTargetFunction();
   }
-  if (!m_target) {
```

Comment:
Doesn't the removal of this check introduce a change for segfault on `m_target->getNumberDomains()`?

Reply:
Good catch! I will revert this change.